jablib/src/main/java/org/jabref/logic/citation/contextextractor/CitationCommentWriter.java

```diff
@@ -0,0 +1,229 @@
+package org.jabref.logic.citation.contextextractor;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import org.jabref.logic.util.strings.StringSimilarity;
+import org.jabref.model.citation.CitationContext;
+import org.jabref.model.entry.BibEntry;
+import org.jabref.model.entry.field.Field;
+import org.jabref.model.entry.field.StandardField;
+import org.jabref.model.entry.field.UserSpecificCommentField;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class CitationCommentWriter {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(CitationCommentWriter.class);
+    private static final String CONTEXT_SEPARATOR = "\n\n";
+    private static final String CONTEXT_FORMAT = "[%s]: %s";
+    private static final double SIMILARITY_THRESHOLD = 0.8;
+
+    private final StringSimilarity stringSimilarity = new StringSimilarity();
+    private final Field commentField;
+    private final String username;
+
+    public CitationCommentWriter() {
+        this.commentField = StandardField.COMMENT;
+        this.username = null;
+    }
+
+    public CitationCommentWriter(String username) {
```

Comment:
Please use preferences. Look around for `getUserAndHost`. Maybe, you need to adapt something there.